internal/grpc/server.go

```diff
@@ -33,17 +34,21 @@ func NewServer(opt *ServerOptions, registrationFn func(s *grpc.Server), wg *sync
 		grpc.KeepaliveParams(opt.KeepaliveParams),
 	}
 
-	if opt.TLSCertificate != nil {
-		log.Debug().Str("addr", opt.Addr).Msg("internal/grpc: serving over TLS")
-		cert := credentials.NewServerTLSFromCert(opt.TLSCertificate)
+	if len(opt.TLSCertificate) == 1 {
+		cert := credentials.NewServerTLSFromCert(&opt.TLSCertificate[0])
 		grpcOpts = append(grpcOpts, grpc.Creds(cert))
-	} else {
-		log.Warn().Str("addr", opt.Addr).Msg("internal/grpc: serving without TLS")
+	} else if !opt.InsecureServer {
+		return nil, errors.New("internal/grpc: unexpected number of certificates")
```

Comment:
Should we retain the insecure warning?

Reply:
I'm cool with it just being part of the Info log on the following line. Thoughts?